### PR TITLE
ISPN-13664 Deadlock on closing of PersistenceManagerImpl

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-13664

### CODE REVIEW

**Comment by danberindei** [comment]
The changes look good, but the main and 13.0.x CI builds did not finish for some reason.
And the 12.1.x CI build had an undelivered exception in PersistenceManagerTest:
https://ci.infinispan.org/job/Infinispan/job/PR-9847/2/testReport/org.infinispan.persistence.manager/PersistenceManagerTest/Undelivered/

**Comment by danberindei** [comment]
`JCacheTwoCachesBasicOpsTest` is failing because `PersistenceManager.enabled == false` and `start()` doesn't initialize the `stores` field, then `PreloadManager.start()` invokes methods that rely on `stores` to be initialized.

**Comment by wburns** [comment]
Tests now pass.

File: core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java [hunk 1]
```diff
@@ -153,6 +155,7 @@
 
    @GuardedBy("lock#readLock")
    private <K, V> NonBlockingStore<K, V> getStoreLocked(Predicate<StoreStatus> predicate) {
+      if (stores == null) return null;
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java:158]: Nitpick: I don't think we put the return on the same line as the if elsewhere, and we usually leave an empty line after it.

```suggestion
      if (stores == null)
         return null;

```

**Review by danberindei — COMMENTED** [review]
Looks good Will, I just have 2 tiny suggestions

File: core/src/test/java/org/infinispan/persistence/manager/PersistenceManagerTest.java [hunk 2]
```diff
@@ -53,9 +57,14 @@ public void testPublishAfterStop() {
...
       persistenceManager.stop();
 
+      ByRef<Throwable> tRef = new ByRef<>(null);
       // The stopped PersistenceManager should never pass the request to the store
       Flowable.fromPublisher(persistenceManager.publishEntries(true, true))
-              .blockingSubscribe(ignore -> fail("shouldn't run"));
+            .blockingSubscribe(ignore -> fail("shouldn't run"), tRef::set);
+
+      Throwable t = tRef.get();
+      assertNotNull(t);
```
[the remark this answers is not in the document]
Comment (wburns) [reply core/src/test/java/org/infinispan/persistence/manager/PersistenceManagerTest.java:66]: Alright :+1:

File: core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java (hunk 1 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java:158]: Sure

**Comment by wburns** [comment]
Updated

File: core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java (hunk 1 above)
Comment (danberindei) [reply core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java]: I didn't mean to add braces as well, but I'll take it :)

**Comment by danberindei** [comment]
Merged, thanks Will!

**Pull request merged**